### PR TITLE
[artf307261] [artf307251] : Modify EventQ plugin to support WEBR

### PR DESCRIPTION
**Pull request by hariharan-cn**
1. Removed EventQ support from this plugin
2. Added plugin support to post data to WEBR
   - Added Notifier and Event listener to send data
   - Modified publisher to support Jenkins pipeline projects
3. Modified Traceability to accept data from TF/WEBR

File changes,
1. BuildNotifier and BuildNotifierDescriptor
   - Event listener for Jenkins Run/Abstract build actions
2. Build Event
   - used to constuct the Build object from Jenkins build
3. PushNotification
   - send/publish data to WEBR rest-end point [webhook url]
4. TraceabiltyAction & TraceabilityActionFactory
   - to project traceability view on Jenkins
5. Resources changes,
   BuildNotifier - config.jelly to accept TF Url, username and Webhook url
   Traceability - included angular-cookies,md5, ui-bootstrap min.js
   PublishEventStep - modified config.jelly to accept Webkhook url
6. tf.css and config.css for css changes
Tests:
1. Removed unused EventQ related tests
2. Added TestPushNotification to test notifier changes
3. Modified TestBuildNotifier and EntryPages to test Webhook url configurations

### CODE REVIEW

File: src/test/java/hudson/plugins/collabnet/orchestrate/TestBuildNotifierDescriptor.java [hunk 1]
```diff
@@ -58,45 +57,33 @@ public void testApplicableToAllProjects() {
...
     /** If you supply something, we'll try to connect to it. */
     @Test
     public void anyStringAllowedForServerUrl() {
-        assertOk(descriptor.doCheckServerUrl("some string"));
+        assertOk(descriptor.doCheckWebhookUrl("some string"));
     }
 
     /** Null source keys are not allowed. */
     @Test
     public void nullSourceKeyNotAllowed() {
```
Comment (BoraAksoy) [inline src/test/java/hudson/plugins/collabnet/orchestrate/TestBuildNotifierDescriptor.java]: this can be removed

File: pom.xml [hunk 2]
```diff
@@ -6,7 +6,7 @@
     <version>1.645</version>
   </parent>
   <artifactId>collabnet</artifactId>
-  <version>2.0.6-SNAPSHOT</version>
+  <version>2.0.6</version>
```
Comment (BoraAksoy) [inline pom.xml]: AFAIK, this is done automatically during the release process

File: src/main/java/hudson/plugins/collabnet/util/Helper.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+     * @param message
+     *            the message to log
+     */
+    public static void markUnstable(Run build,
+                              PrintStream consoleLogger, String message, String className) {
+        if(consoleLogger !=null ){
+            log(message, consoleLogger);
+            Logger logger = Logger.getLogger(className);
+            logger.warning(message);
+            build.setResult(Result.UNSTABLE);
```
Comment (BoraAksoy) [inline src/main/java/hudson/plugins/collabnet/util/Helper.java]: Is it intentional that you are marking the build as **UNSTABLE** only if  `consoleLogger ` is not null.

File: src/main/java/hudson/plugins/collabnet/util/Helper.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+public class Helper {
+
+    /** Prefix for messages appearing in the console log, for readability */
+    private static String LOG_MESSAGE_PREFIX = "TeamForge WEBR Build Notifier - ";
+
+    static Logger logger = Logger.getLogger("hudson.plugins.collab.orchestrate");
+
+    public static String getToken(URL ctfUrl, String ctfUserName, String ctfPassword) throws IOException {
+        String end_point = ctfUrl.toString()+"/oauth/auth/token";
+        HttpClient httpclient = new DefaultHttpClient();
```
Comment (BoraAksoy) [inline src/main/java/hudson/plugins/collabnet/util/Helper.java]: Please make sure HttpClient/Connections are closed/disposed properly. Also, CollabNet plugin allows ignoring SSL warnings/errors, however this implementation does not take the setting into account. See `CNFormFieldValidator.java` for an example. This should apply to all HTTP connections.

File: src/main/java/hudson/plugins/collabnet/orchestrate/PushNotification.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        }
+    }
+
+    public String getPayload(Run build, TaskListener listener, String status, boolean excludeCommitInfo) throws
+            IOException {
+        return BuildEvent.constructJson(build, listener, status, excludeCommitInfo).toString();
+    }
+
+    private int send(String url, String buildData) throws IOException {
+        CloseableHttpClient client = HttpClients.createDefault();
```
Comment (BoraAksoy) [inline src/main/java/hudson/plugins/collabnet/orchestrate/PushNotification.java]: The setting to ignore SSL warnings/errors is not taken into account

File: src/main/java/jenkins/plugins/collabnet/steps/PublishEventStep.java [hunk 6]
```diff
@@ -93,8 +83,8 @@ public StepExecution start(StepContext context) throws Exception {
      *
      * @return the server URL
      */
-    public String getServerUrl() {
-        return serverUrl;
+    public String getWebhookUrl() {
+        return webhookUrl;
     }
 
     public String getCredentialsId() {
```
Comment (BoraAksoy) [inline src/main/java/jenkins/plugins/collabnet/steps/PublishEventStep.java]: Are credentials needed for the new step?

File: src/test/java/hudson/plugins/collabnet/orchestrate/TestBuildNotifierDescriptor.java (hunk 1 above)
Comment (hariharan-cn) [reply src/test/java/hudson/plugins/collabnet/orchestrate/TestBuildNotifierDescriptor.java]: sure, will be removed.

File: pom.xml (hunk 2 above)
Comment (hariharan-cn) [reply pom.xml]: OK

File: src/main/java/hudson/plugins/collabnet/util/Helper.java (hunk 3 above)
Comment (hariharan-cn) [reply src/main/java/hudson/plugins/collabnet/util/Helper.java]: Yes, it's intentional. making build unstable when SCM not configured. consoleLogger null check used for Tests.

File: src/main/java/hudson/plugins/collabnet/util/Helper.java (hunk 4 above)
Comment (hariharan-cn) [reply src/main/java/hudson/plugins/collabnet/util/Helper.java]: Sure, will update this.

File: src/main/java/hudson/plugins/collabnet/orchestrate/PushNotification.java (hunk 5 above)
Comment (hariharan-cn) [reply src/main/java/hudson/plugins/collabnet/orchestrate/PushNotification.java]: Sure, will update this.

File: src/main/java/jenkins/plugins/collabnet/steps/PublishEventStep.java (hunk 6 above)
Comment (hariharan-cn) [reply src/main/java/jenkins/plugins/collabnet/steps/PublishEventStep.java]: Yeah, credentials are not required at least for now. But if we introduce traceability view for pipeline projects then credentials are required. will comment credentials for now.

**Comment by hariharan-cn** [comment]
Agree, plugin version 2.0.5 will still support EventQ based integration. 
From version 2.0.6, we should use WEBR based integrations and won't support EventQ.

**Comment by BoraAksoy** [comment]
I meant to support earlier CTF versions, i.e. keep supporting EventQ integration along with WEBR integration in this plugin. IOW, we need to have a grace period before removing the EVENTQ support. So that we won't need to create "release" branches for users of earlier CTF versions

**Comment by BoraAksoy** [comment]
closing this as it is duplicate and https://github.com/jenkinsci/collabnet-plugin/pull/24 has already been merged.

**Pull request closed without merging**